src/providers/moduleProvider.ts

```diff
@@ -0,0 +1,147 @@
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { ExecuteCommandParams, ExecuteCommandRequest } from 'vscode-languageclient';
+import { Utils } from 'vscode-uri';
+import { ClientHandler } from '../clientHandler';
+import { getActiveTextEditor } from '../vscodeUtils';
+
+class ModuleCall extends vscode.TreeItem {
+  constructor(
+    public label: string,
+    public sourceAddr: string,
+    public version: string,
+    public sourceType: string,
+    public docsLink: string,
+    public terraformIcon: string,
+    public readonly children: ModuleCall[],
+  ) {
+    super(
+      label,
+      children.length >= 1 ? vscode.TreeItemCollapsibleState.Collapsed : vscode.TreeItemCollapsibleState.None,
+    );
+
+    this.description = this.version ? `${this.version}` : '';
+
+    if (this.version === undefined) {
+      this.tooltip = `${this.sourceAddr}`;
+    } else {
+      this.tooltip = `${this.sourceAddr}@${this.version}`;
+    }
+  }
+
+  iconPath = this.getIcon(this.sourceType);
+
+  getIcon(type: string) {
+    switch (type) {
+      case 'tfregistry':
+        return {
+          light: this.terraformIcon,
+          dark: this.terraformIcon,
+        };
+      case 'local':
+        return new vscode.ThemeIcon('symbol-folder');
+      case 'github':
+        return new vscode.ThemeIcon('github');
+      case 'gitlab':
+        return new vscode.ThemeIcon('gitlab');
```

Comment:
```suggestion
```
As mentioned in my review of the LS PR Terraform doesn't treat GitLab differently than any other git repository, so I _think_ we probably shouldn't either, if only for consistency.